docs/software-modules.md

```diff
@@ -0,0 +1,129 @@
+Software modules for Slurm clusters
+===================================
+
+Slurm clusters will frequently provide a bare-metal development environment in the form of "Environment Modules".
+Enviroment Modules provide a convenient way to dynamically change the user's environment variables, such as `PATH` or `LD_LIBRARY_PATH`, to enable use of software installed in different locations.
+This makes it easy to install multiple versions of the same software package and dynamically switch between them, such as different versions of CUDA or OpenMPI.
+
+DeepOps installs the Lmod tool for managing your Environment Modules.
+The [Lmod documentation](https://lmod.readthedocs.io/en/latest/) provides more information on using Lmod and writing Modules.
+
+DeepOps also provides tooling for managing Environment Modules with two different frameworks: EasyBuild and Spack.
+These tools provide a selection of tested build recipes for common development tools and scientific software packages,
+making it easier to install software without a tedious manual build process.
+
+
+## Installing Spack
+
+[Spack](https://spack.io/) is a package manager for supercomputers, Linux, and MacOS which makes installing scientific software easy.
+
+To install Spack using DeepOps, you can run:
+
+```bash
+ansible-playbook -l slurm-cluster playbooks/spack-modules.yml
```

Comment:
Great documentation! I think we either need to call out running nfs-server/nfs-client playbooks since they're required and we don't currently run them automatically, or run those playbooks by default somewhere in the slurm-cluster process.

Reply:
Agree! I think I'll probably put the NFS playbooks by default in slurm-cluster, but that's going to be a separate PR because I want to also add some NFS docs at the same time.

Reply:
When you do that you might want to clean up the nfs variables. They're currently in all.yml and in slurm-cluster.yml. I ran into some situations where this caused a bit of confusion because people were updating the all.yml and it was not clear that those values are only used for stand-alone or kubernetes (not slurm-cluster).

Reply:
That's a pretty fundamental Ansible concept though. I think it makes sense that we have NFS config which is not Slurm related, but also default NFS config that is required for Slurm. Maybe just make sure we have clear comments in the respective files?